### PR TITLE
admin2: Add bans

### LINKED ISSUE #151: admin2: add extended ban features
**Is your feature request related to a problem? Please describe.**
admin2 currently has no feature parity with admin1 when it comes to ban features

**Describe the solution you'd like**
Make sure ban-related features that are present in admin1 will also be in admin2

**Describe alternatives you've considered**
The current ban-related features in admin2 aren't suitable for modern-day use. For example, there's only support for IP banning and not serial banning. Besides that, the set of features are very limited; see below at context. We really need a rewrite of ban features in admin2 overall, no alternatives.

**Additional context**

Current situation:
- There is only 1 way to add a ban: select player from list. You cannot offline-ban or add manual bans in the "Bans" tab.
- The ban you can add from playerlist is limited to an IP ban.
- The "Bans" tab offers an "Unban" button for IP bans. That's it.

So we currently got a minimal, outdated set of ban-related features.

Required situation:
- You can add bans both from player list and the "Bans" tab.
- Both serial and IP bans are supported, online ban (from player list) and "Bans" tab should largely share the same ban customisation window

The ban customisation window should include:
* For online bans:
- Ban reason
- Ban type (IP / serial)
- Duration (custom mins/days/hours and predefined values like in admin1)

* For offline bans/from the "Bans" tab:
- "Ban IP" and "Ban serial" buttons like in admin1 (or as simplification over admin1, merge them into "Add ban" in admin2)
- Duration (custom mins/days/hours)
- Fields to enter player nick and ban reason

Implementing this list would result in feature parity with admin1 and suitability for modern-day usage in regards to banning.
To further explain, I will add screenshots of current ban-related features supported by admin1 that were described:

ban1

ban2

### PR DESCRIPTION
**Pull request by iDannz1**
This is still in progress and I don't know exactly when it should be ready for review...
Don't expect it to work yet, most of the bans stuff in admin2 is still broken... I should update it as soon as I have free time again. If you want to help, PRs in my fork are welcome 👀

Once merged this should close #151.

### CODE REVIEW

**Comment by Fernando-A-Rocha** [comment]
Picking this work and I intend to finish the work. Using #151 as reference of what's expected.

File: [admin]/admin2/client/widgets/admin_ban.lua [hunk 1]
```diff
@@ -6,91 +6,280 @@
...
+        local sW, sH = guiGetScreenSize()
+        local w, h = 357, 282
+
+        aHandleBans.Form = guiCreateWindow((sW - w) / 2, (sH - h) / 2, w, h, 'ban player', false)
+        guiWindowSetSizable(aHandleBans.Form, false)
+        guiSetAlpha(aHandleBans.Form, 1)
+
+        local y = 25
+
+        aHandleBans.NameAndReasonlabel = guiCreateLabel(28, y, 301, 20, 'Enter the ban name and reason', false, aHandleBans.Form)
```
Comment (Haxardous) [inline [admin]/admin2/client/widgets/admin_ban.lua:37]: would be great to rephrase the label's sentence

**Comment by Dutchman101** [comment]
I expect this to be replaced by PR #437

**Comment by Dutchman101** [comment]
Closing in favour of #437

**Pull request closed without merging**